reconcile/openshift_base.py

```diff
@@ -185,8 +194,12 @@ def apply(dry_run, oc_map, cluster, namespace, resource_type, resource):
         # do not skip if this is a cluster scoped integration
         if namespace != 'cluster' and not oc.project_exists(namespace):
             msg = f"[{cluster}/{namespace}] namespace does not exist (yet)"
-            logging.warning(msg)
-            return
+            if wait_for_namespace:
+                logging.info(msg)
+                wait_for_namespace_exists(oc, namespace)
```

Comment:
Does this mean that the whole integration with `wait_for_namespace`  is going to fail hard in case of one of the namespaces do not exist?

Reply:
and this will only apply to openshift-saas-deploy. we shouldn't enable it for integrations running continuously (although that will not be super harmful).

Reply:
Sure, but even for openshift-saas-deploy... We move from warning and moving on to failing hard the whole integration. Why can't we keep the old behaviour with the retry mechanism?

Reply:
we want openshift-saas-deploy to fail hard if it was not able to deploy. unlike other integrations, this will only run once, and we'd like to know that it failed. does that make sense?

Reply:
If that's the case, if it's possible I would control the behaviour in the integration, not in a library that's consumed by lots of integrations

Reply:
not possible with the current flow. and we may have a case in the future when we want to have this applied to other integrations as well...

Reply:
d'oh... then a nice comment about the intentions and implications of using `wait_for_namespace` in the function declaration for those that actually go and have a look would be nice 😍

Reply:
added documentation to the `realize_data` method. thanks!